core/arch/arm/encode.c

```diff
@@ -2766,7 +2766,7 @@ instr_encode_arch(dcontext_t *dcontext, instr_t *instr, byte *copy_pc, byte *fin
     }
 
     decode_info_init_for_instr(&di, instr);
-    di.opcode = instr_get_opcode(instr);
+    di.opcode = instr->opcode;
```

Comment:
I think there are perhaps two general, cross-architecture things here which I don't want to address in this commit:

* `instr_get_opcode` looks like a harmless accessor but may invoke the (full) decoder.

* `OP_UNDECODED` might mean that we haven't got round to decoding this instruction, or it might mean that we have tried and found that the encoding does not correspond to any known instruction (except on AArch64 where the decoder always succeeds). Perhaps this distinction is well-defined somehow, but it's not clear to me, at least.